app/controllers/ReviewApplicationController.scala

```diff
@@ -49,8 +50,8 @@ abstract class ReviewApplicationController(applicationClient: ApplicationClient)
   def submit = CSRSecureAppAction(ReviewRole) { implicit request =>
     implicit user =>
       applicationClient.updateReview(user.application.applicationId).flatMap { _ =>
-        updateProgress() { u =>
-          if (StartQuestionnaireRole.isAuthorized(u) || QuestionnaireInProgressRole.isAuthorized(u)) {
+        updateProgress() { usr =>
+          if (StartQuestionnaireRole.isAuthorized(usr) && QuestionnaireInProgressRole.isAuthorized(usr)) {
```

Comment:
Just QuestionnaireInProgressRole is compared here as the StartQuestionnaireRole now does not include any state checks on the questionnaire as such.